### PR TITLE
Validate OPTIONS argument

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Ensure that the argument to an AQL OPTIONS clause is always an object which does not contain any dynamic (run-time) values. 
Previously, this was only enforced for traversal options and options for data-modification queries. This change extends the check to all occurrences of OPTIONS.

- [x] :hankey: Bugfix 
- [ ] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [x] Backports required for: 3.7

### Testing & Verification

- [x] This change is already covered by existing tests, such as *shell_server_aql*.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11800/

### CODE REVIEW

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11803/

**Review by danielhlarkin — APPROVED** [review]

**Review by Simran-B — APPROVED** [review]
Now raises an error:

> Explain: 'OPTIONS' have to be readable at query compile time near 'RETURN v' at position 3:1 (while parsing)

File: arangod/Aql/grammar.y [hunk 1]
```diff
@@ -65,6 +60,17 @@ void Aqlerror(YYLTYPE* locp,
...
 
 namespace {
+void validateOptions(Parser* parser, AstNode const* node,
+                     int line, int column) {
+  TRI_ASSERT(node != nullptr);
+  if (!node->isObject()) {
+    parser->registerParseError(TRI_ERROR_QUERY_PARSE, "'OPTIONS' have to be an object", line, column);
+  }
+  if (!node->isConstant()) {
+    parser->registerParseError(TRI_ERROR_QUERY_COMPILE_TIME_OPTIONS, "'OPTIONS' have to be readable at query compile time", line, column);
```
Comment (Simran-B) [inline arangod/Aql/grammar.y:70]: Improve wording?

```suggestion
    parser->registerParseError(TRI_ERROR_QUERY_COMPILE_TIME_OPTIONS, "'OPTIONS' have to be known at query compile time", line, column);
```

Comment (jsteemann) [reply arangod/Aql/grammar.y:70]: Happy with that. I just copied the old error message, but will now apply your changes.

**Pull request merged**